### ISSUE TITLE
CUBLAS_STATUS_EXECUTION_FAILED when invoke torch.linalg.lstsq to handle large input tensor

### ISSUE THREAD

**Issue by CielAl** [pytorch/pytorch#117507]
### 🐛 Describe the bug

Hi,

I’ve been using torch.linalg.lstsq recently but it appears that a cuda run time error may occur on GPU when the input size is relatively large, and I am not sure whether the error is indeed due to the large matrix size or due to CUDA/torch versions, since there was no error with CPU tensor of the same size, or GPU tensor with smaller size.
The error is

```RuntimeError: CUDA error: CUBLAS_STATUS_EXECUTION_FAILED when calling cublasStrsmBatched( handle, side, uplo, trans, diag, m, n, alpha, A, lda, B, ldb, batchCount)```

A minimum working example to reproduce the error is here:

```python
import torch
a = torch.rand(1, 3, 2)
b = torch.rand(1, 3, 2500*2500) # 1 x 3 x 6250000
# no error
torch.linalg.lstsq(a, b)
# error
torch.linalg.lstsq(a.cuda(), b.cuda())
# no error if making b smaller
torch.linalg.lstsq(a.cuda(), b[:, :, :65536].cuda())
```

### Versions

Collecting environment information…
PyTorch version: 1.13.1
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Pro
GCC version: (Rev5, Built by MSYS2 project) 5.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.9 | packaged by Anaconda, Inc. | (main, Mar 8 2023, 10:42:25) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19045-SP0
Is CUDA available: True
CUDA runtime version: 11.7.64
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA TITAN RTX
GPU 1: NVIDIA TITAN RTX

Nvidia driver version: 536.23
[conda] mkl_random 1.2.2 py310h4ed8f06_0
[conda] numpy 1.23.5 py310h60c9a35_0
[conda] numpy-base 1.23.5 py310h04254f7_0
[conda] pytorch 1.13.1 py3.10_cuda11.7_cudnn8_0 pytorch
[conda] pytorch-cuda 11.7 h67b0de4_1 pytorch
[conda] pytorch-lightning 2.0.3 py310haa95532_0
[conda] pytorch-mutex 1.0 cuda pytorch
[conda] torchaudio 0.13.1 pypi_0 pypi
[conda] torchmetrics 1.0.3 pyhd8ed1ab_0 conda-forge
[conda] torchnet 0.0.4 pypi_0 pypi
[conda] torchvision 0.15.2 cpu_py310h7187fe4_0

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht @iremyux @Blackhex @cristianPanaite @ptrblck @jianyuh @nikitaved @pearu @mruberry @walterddr @xwang233 @Lezcano

**Comment by malfet** [comment]
Is there anything that prevents you from updating to PyTorch-2.1+?

**Comment by CielAl** [comment]
> Is there anything that prevents you from updating to PyTorch-2.1+?

Hi. The issue was encountered in a project-specific environment, but here goes a similar issue with slight different error message in 2.1+ from the [forum](https://discuss.pytorch.org/t/cuda-error-over-torch-linalg-lstsq-with-large-input-tensor-size/195317/2):

```python
import torch
print (torch.__version__)
print (torch.version.cuda)
print (torch.cuda.get_device_properties (0))

_ = torch.manual_seed (2024)

a = torch.rand(1, 3, 2)
b = torch.rand(1, 3, 2500*2500) # 1 x 3 x 6250000
torch.linalg.lstsq(a.cuda(), b.cuda())
```
Error:
```
2.1.2
11.8
_CudaDeviceProperties(name='GeForce GTX 1050 Ti', major=6, minor=1, total_memory=4040MB, multi_processor_count=6)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 14, in <module>
RuntimeError: CUDA error: CUBLAS_STATUS_NOT_SUPPORTED when calling `cublasStrsmBatched( handle, side, uplo, trans, diag, m, n, alpha, A, lda, B, ldb, batchCount)`
```

**Comment by malfet** [comment]
Interesting, this looks GPU or Platform specific, as the abovementioned code works fine for me on Linux:
```
>>> print (torch.__version__)
2.1.2+cu121
>>> print (torch.version.cuda)
12.1
>>> print (torch.cuda.get_device_properties (0))
_CudaDeviceProperties(name='NVIDIA A100-SXM4-40GB', major=8, minor=0, total_memory=40513MB, multi_processor_count=108)
>>> 
>>> _ = torch.manual_seed (2024)
>>> 
>>> a = torch.rand(1, 3, 2)
>>> b = torch.rand(1, 3, 2500*2500) # 1 x 3 x 6250000
>>> torch.linalg.lstsq(a.cuda(), b.cuda())
torch.return_types.linalg_lstsq(
solution=tensor([[[ 8.2919e-01,  7.2121e-01,  2.6185e-01,  ...,  1.3793e-01,
           7.0426e-01,  9.7874e-01],
         [-9.9643e-05,  5.9296e-01,  4.0041e-01,  ...,  9.2249e-01,
           9.8641e-02, -1.2075e-01]]], device='cuda:0'),
residuals=tensor([[0.3286, 0.4231, 0.0011,  ..., 0.1359, 0.2573, 0.2141]],
       device='cuda:0'),
rank=tensor([], device='cuda:0', dtype=torch.int64),
singular_values=tensor([], device='cuda:0'))
```

**Comment by CielAl** [comment]
> Interesting, this looks GPU or Platform specific, as the abovementioned code works fine for me on Linux:
> 
> ```
> >>> print (torch.__version__)
> 2.1.2+cu121
> >>> print (torch.version.cuda)
> 12.1
> >>> print (torch.cuda.get_device_properties (0))
> _CudaDeviceProperties(name='NVIDIA A100-SXM4-40GB', major=8, minor=0, total_memory=40513MB, multi_processor_count=108)
> >>> 
> >>> _ = torch.manual_seed (2024)
> >>> 
> >>> a = torch.rand(1, 3, 2)
> >>> b = torch.rand(1, 3, 2500*2500) # 1 x 3 x 6250000
> >>> torch.linalg.lstsq(a.cuda(), b.cuda())
> torch.return_types.linalg_lstsq(
> solution=tensor([[[ 8.2919e-01,  7.2121e-01,  2.6185e-01,  ...,  1.3793e-01,
>            7.0426e-01,  9.7874e-01],
>          [-9.9643e-05,  5.9296e-01,  4.0041e-01,  ...,  9.2249e-01,
>            9.8641e-02, -1.2075e-01]]], device='cuda:0'),
> residuals=tensor([[0.3286, 0.4231, 0.0011,  ..., 0.1359, 0.2573, 0.2141]],
>        device='cuda:0'),
> rank=tensor([], device='cuda:0', dtype=torch.int64),
> singular_values=tensor([], device='cuda:0'))
> ```

Since it did not happen to CPU tensors or GPU tensor with smaller size - would the size of the available VRAM be a factor here?

**Comment by lezcano** [comment]
Good to see that this is fixed in CUDA 12.1 tho :)

**Comment by CielAl** [comment]
> Good to see that this is fixed in CUDA 12.1 tho :)

Thanks for the feedback! Good to know that it will be fixed in future.
